src/Optimizations/itensor_network.jl

```diff
@@ -0,0 +1,25 @@
+using ChainRulesCore
+using ..ITensorNetworks
+using ..ITensorNetworks: split_network, default_projector_center
+
+function ChainRulesCore.rrule(
+  ::typeof(split_network),
```

Comment:
Since `split_network` only involves priming right now, we could make the reverse rule something like:
```julia
broadcast(t -> replaceprime(t, 1 => 0),  dtn_split)
```
just to make the implementation simpler and not depend on Index `id`s. In general I think it is not good practice to write code that make use of Index ids, since multiple indices in the network can have the same ids but different tags or prime levels that distinguish them.

Reply:
done